selfservice/strategy/oidc/strategy_login.go

```diff
@@ -432,18 +454,23 @@ func (s *Strategy) PopulateLoginMethodIdentifierFirstCredentials(r *http.Request
 		}
 
 		// We found no credentials. We remove all the providers and tell the strategy that we found nothing.
-		f.GetUI().UnsetNode("provider")
+		s.removeProviders(conf, f)
```

Comment:
`UnsetNode` would delete all provider fields, which doesn't make sense. Instead, removeProviders now only removes providers coming from this strategy's configuration.